neqo-bin/src/client/http09.rs

```diff
@@ -85,6 +86,28 @@ impl super::Handler for Handler<'_> {
                         self.download_urls(client);
                     }
                 }
+                ConnectionEvent::StateChange(State::Confirmed) => {
+                    if let Some((local_port, migration_addr)) = &self.migration {
```

Comment:
Is there any reason why we would want to keep trying?  It seems like you could use this with no loss of generality, because we only ever generate ONE State::Confirmed event for a connection.

```suggestion
                    if let Some((local_port, migration_addr)) = self.migration.take() {
```

That leads to a few minor changes below...